spec_macros/src/derive_spec/spec_enum.rs

```diff
@@ -0,0 +1,524 @@
+
+                    spec_variants.push(quote! {
+                        #(#combined_attrs )*
+                        #variant_ident ( #(#new_variant),* )
+                    });
+
+                    // create a vector field_<i> for each unnamed field
+                    let bindings = (0..unnamed.len())
+                        .map(|i| format_ident!("field_{i}"))
+                        .collect::<Vec<_>>();
+
+                    try_from_variants.push(quote! {
+                                #orig_name::#variant_ident( #(#bindings),* ) => #spec_name::#variant_ident( #(#try_fields),* )
+                            });
+
+                    from_variants.push(quote! {
+                                #spec_name::#variant_ident( #(#bindings),* ) => #orig_name::#variant_ident( #(#from_fields),* )
+                            });
+                };
+            }
+
+            Fields::Unit => {
+                spec_variants.push(quote! {
+                    #(#combined_attrs )*
+                    #variant_ident
+                });
+
+                try_from_variants.push(quote! {
+                    #orig_name::#variant_ident => #spec_name::#variant_ident
+                });
+
+                from_variants.push(quote! {
+                    #spec_name::#variant_ident => #orig_name::#variant_ident
+                });
+            }
+        };
+    }
+
+    let spec_enum = quote! {
+        #(#type_attrs )*
+        #vis enum #spec_name {
+            #(#spec_variants,)*
+        }
+    };
+
+    let try_from_impl = quote! {
+        impl std::convert::TryFrom<#orig_name> for #spec_name {
+            type Error = String;
+
+            fn try_from(orig: #orig_name) -> Result<Self, Self::Error> {
+                Ok(match orig {
+                    #(#try_from_variants),*
+                })
+            }
+        }
+    };
+
+    let from_impl = quote! {
+        impl From<#spec_name> for #orig_name {
+            fn from(original: #spec_name) -> Self {
+                match original {
+                    #(#from_variants),*
+                }
+            }
+        }
+    };
+
+    Ok(DerivedSpec {
+        obj: spec_enum,
+        try_from_impl,
+        from_impl,
+    })
+}
+
+//////////////////////////////////////////////////////////////////////////////
+//                 ########  #######    #########  #########                //
+//                    ##     ##        ##             ##                    //
+//                    ##     #####     #########      ##                    //
+//                    ##     ##                ##     ##                    //
+//                    ##     #######   #########      ##                    //
+//////////////////////////////////////////////////////////////////////////////
+
+#[cfg(test)]
+mod tests {
+
+    use proc_macro2::TokenStream;
+    use quote::{format_ident, quote};
+    use syn::{Visibility, parse_quote};
+
+    use super::derive_spec_enum;
```

Comment:
Same logic as before with the imports
```suggestion
    use super::derive_spec_enum;
    
    use proc_macro2::TokenStream;
    use quote::{format_ident, quote};
    use syn::{Visibility, parse_quote};
```